internal/lexer/token.go

```diff
@@ -0,0 +1,51 @@
+package lexer
+
+type TokenType int
+
+const (
+	EOF TokenType = iota
+	IDENTITY
+	NUMBER
```

Comment:
Why do we need a number token? I think that all our values should be strings. What extra value do we get from differentiating between a string and a number?